manifests/monitor/monitor-configMap.yaml

```diff
@@ -0,0 +1,190 @@
+        action: replace
+        target_label: kubernetes_namespace
+      - source_labels: [__meta_kubernetes_pod_node_name]
+        action: replace
+        target_label: kubernetes_node
+      - source_labels: [__meta_kubernetes_pod_ip]
+        action: replace
+        target_label: kubernetes_pod_ip
+      - source_labels: [__meta_kubernetes_pod_name]
+        action: replace
+        target_label: instance
+      - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance]
+        action: replace
+        target_label: cluster
+    - job_name: 'tikv'
+      scrape_interval: 15s
+      honor_labels: true
+      kubernetes_sd_configs:
+      - role: pod
+        namespaces:
+          names:
+          - default
+      scheme: http
+      tls_config:
+        insecure_skip_verify: true
+      # TiKV doesn't support scheme https for now.
+      # And we should fix it after TiKV fix this issue: https://github.com/tikv/tikv/issues/5340
+      #
+      # scheme: http
+      # tls_config:
+      #   insecure_skip_verify: true
+      #
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance]
+        action: keep
+        regex: test
+      - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_component]
+        action: keep
+        regex: tikv
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
+        action: keep
+        regex: true
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
+        action: replace
+        target_label: __metrics_path__
+        regex: (.+)
+      - source_labels: [__meta_kubernetes_pod_name, __meta_kubernetes_pod_label_app_kubernetes_io_instance,
+          __meta_kubernetes_pod_annotation_prometheus_io_port]
+        regex: (.+);(.+);(.+)
+        target_label: __address__
+        replacement: $1.$2-tikv-peer:$3
+        action: replace
+      - source_labels: [__meta_kubernetes_namespace]
+        action: replace
+        target_label: kubernetes_namespace
+      - source_labels: [__meta_kubernetes_pod_node_name]
+        action: replace
+        target_label: kubernetes_node
+      - source_labels: [__meta_kubernetes_pod_ip]
+        action: replace
+        target_label: kubernetes_pod_ip
+      - source_labels: [__meta_kubernetes_pod_name]
+        action: replace
+        target_label: instance
+      - source_labels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance]
+        action: replace
+        target_label: cluster
+    rule_files:
+      - '/prometheus-rules/rules/*.rules.yml'
+
+  dashboard-config: |-
+    {
+        "apiVersion": 1,
+        "providers": [
+            {
+                "folder": "",
+                "name": "0",
+                "options": {
+                    "path": "/grafana-dashboard-definitions/tidb"
+                },
+                "orgId": 1,
+                "type": "file"
+            }
+        ]
+    }
+  start-command: |-
```

Comment:
IMO the `start-command` is not that flexible and scalable. A possible way is to do something like Prometheus-operator https://github.com/coreos/prometheus-operator/blob/87a7bea9f028c62396bef665f2887df326fa97f7/pkg/apis/monitoring/v1/types.go#L244. We can have a `containers` field in the tidb monitor spec. Then in the real generating phase, we do a patch to containers like [here](https://github.com/coreos/prometheus-operator/blob/87a7bea9f028c62396bef665f2887df326fa97f7/pkg/prometheus/statefulset.go#L889).

This approach is more recommended since it can help specify something other than the start commands for Prometheus, like env vars, volumes, etc

Reply:
I have only support command field patch in container, other field patch I think is not important now.